### PR TITLE
Make the base branch generic

### PR DESCRIPTION
**Pull request by fmarco76**
In case of commits to different branches the rebase has to be performed against the base branch of the PR. However, the code analysis execute in a workflow_run event and it has no information of the PR base, as in the [docuemtnation](https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#webhook-payload-example-51).

The available options are:

    to store the base branch during the original PR event and retrieve this information during the workflow run
    to retrieve the base branch using some javascript code interacting with GitHab rest API.

The former is followed here.
Of course, in order to work Sonarcloud in all the branch these have to include the relative workflows but this is required in any case to perform the analysis of them.

### CODE REVIEW

**Review by edewata — APPROVED** [review]
LGTM. Some of the tests are failing, but I don't think they're caused by this PR. The DS test failure should be fixed by PR #25.

File: .github/workflows/sonarcloud-pull.yml [hunk 1]
```diff
@@ -4,13 +4,14 @@ on:
     workflows: ["Code Analysis"]
     types:
       - completed
-
 jobs:
+
```
Comment (edewata) [inline .github/workflows/sonarcloud-pull.yml]: Just one thing though, I think we should restore the original spacing.

Comment (fmarco76) [reply .github/workflows/sonarcloud-pull.yml]: Yes,  it is better!

**Review by ckelleyRH — APPROVED** [review]
LGTM!

**Pull request merged**